Automattic/portkey-android
# Remove Ruby (replacing it with the `configure` gradle plugin)

This PR resolves a Ruby security warning by removing Ruby from the project entirely and replacing it with the new  `configure` gradle plugin, which we're rolling out to Android projects.

**To Test:**
- Ensure all tests pass in CI and locally – before running them locally, run `./gradlew applyConfiguration` to update the local secrets
- Download the installable build and ensure it works as expected compared to other PRs

@mzorz or @aforcier – would one of you be able to check the code changes and make sure the installable build works as expected?

@oguzkocer – could you take a pass over the gradle configuration? The fact that Android doesn't wrap its configuration in a task, but instead has to have everything present during the script evaluation pass means that we need a bunch of early returns in order for `./gradlew applyConfiguration` to have a chance to copy `gradle.properties` into position before errors are emitted. If you know of another solution I'm open to it 😅
